CleverRaven/Cataclysm-DDA
# parameterized field base camp version 2

#### Summary

None



#### Purpose of change

Main purpose:
- Make use of the ability to use palette parameters in mapgen/recipes to reduce the size of the JSON for field base camp version 2.

Side purposes:
- Remove bed/cooking construction of the first building as prerequisite for further construction.
- Add the option of making makeshift beds (made easy by the palette parameters).



#### Describe the solution

Massaging the construction recipes into palette parameter based ones.



#### Describe alternatives you've considered

Leave the new functionality implemented for this purpose unused.



#### Testing

The main testing was performed while testing the functionality (#58180), but a quick sanity check constructing the whole base camp with "random" materials for different parts was done before producing this PR.

- Verify that all construction materials actually can be used (but possibly not for every individual segment, although that probably was done before the PR went into the deep freezer last year).
- Verify that all 3 first building cooking facilities can be constructed.
- Verify that all 3 bed options can be constructed. Every option for every position has not been tested, though, but rather all permutations for the first room and "random" ones for the rest of them.



#### Additional context
